File: java/Q19.java
public class RemoveNthNodeFromEndOfList {
    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
        }

    public ListNode remove (ListNode head, int n){
        ListNode dummy_head = new ListNode(0);
        dummy_head.next = head; 
        ListNode slow = dummy_head;
        ListNode fast = dummy_head; 

        for(int i = 1 ; i<= n+1; i++){
            fast = fast.next; 
        }

        while(fast != null){
            slow = slow.next; 
            fast = fast.next; 
        }

        slow.next = slow.next.next; 


        return dummy_head.next;
    }
}

File: python/Q19.py
#19. Remove Nth Node from End Of List


class ListNode:
    def __init__(self) -> None:
        self.val = self
        self.next = next 

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n:int) -> Optional[ListNode]:

        fast, slow = head, head

        for _ in range(n):
            fast = fast.next

        if not fast:
            return head.next
        
        while fast.next:
            slow = slow.next
            fast = fast.next
        

        slow.next = slow.next.next

        return head